app/src/editor/canvas/components/CanvasController/useCanvasLoadCallback.js

```diff
@@ -13,6 +13,9 @@ export default function useCanvasLoadCallback() {
     return useCallback(async (canvasId) => (
         wrap(async () => {
             const canvas = await services.loadRelevantCanvas({ id: canvasId })
+            // Get permissions and save them temporarily to canvas
+            const permissions = await services.getCanvasPermissions({ id: canvasId })
+            canvas.permissions = permissions.data
```

Comment:
note `loadRelevantCanvas` can actually return a different canvas id e.g. if there's a non-unlinked historical canvas, it'll return that canvas.

Reply:
So using `id: canvas.id` for permissions call should do the trick?

Reply:
correct